src/frontend/src/pages/index.js

```diff
@@ -14,33 +41,42 @@ export default function IndexPage() {
   const { telescopeUrl } = useSiteMetaData();
   const savedCallback = useRef();
 
+  // Pagination
+  const [nextPageLink, setNextPageLink] = useState(`/posts?page=${numPages}`);
+  const [loading, setLoading] = useState(false);
+  const classes = useStyles();
+
   useEffect(() => {
-    async function getPosts(pageNum = 1) {
+    async function getPosts() {
       try {
-        const res = await fetch(`${telescopeUrl}/posts?page=${pageNum}`);
+        setLoading(true);
+        const res = await fetch(`${telescopeUrl}${nextPageLink}`);
+
         if (!res.ok) {
           throw new Error(res.statusText);
         }
-        const postsUrls = await res.json();
+
+        const postUrls = await res.json();
+        const links = parse(res.headers.get('Link'));
         const postsData = await Promise.all(
-          postsUrls.map(async ({ url }) => {
-            const tmp = await fetch(`${telescopeUrl}${url}`);
-            if (!tmp.ok) {
-              throw new Error(tmp.statusText);
-            }
-            const post = await tmp.json();
-            return post;
-          })
+          postUrls.map(({ url }) => fetch(`${telescopeUrl}${url}`).then((resp) => resp.json()))
         );
+
         setPosts([...posts, ...postsData]);
-        setNumPages(numPages + 1);
+        setNextPageLink(links.next.url);
+        setLoading(false);
```

Comment:
Do this in the error case below in the `catch` too, or it will spin forever with an error

Reply:
fixed